code/src/App.js

```diff
@@ -1,9 +1,28 @@
-import React from 'react'
+import React, { useState } from 'react'
+import musicData from './data.json'
+import { Header } from './components/Header'
+import { Album } from './components/Album'
+import { EmptyState } from './components/EmptyState'
 
 export const App = () => {
+  const [data, setData] = useState(musicData.albums.items);
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
+    <>
+      <Header toChild={data} sendToParent={setData} />
+      <section className="albums">
+        {(!data.length)
+          ? <EmptyState message="No music found.." />
+          : data.map((item) => {
+            return (
+              <div className="album" key={item.id}>
+                <Album
+                  key={item.id}
```

Comment:
The `key` attribute is considered as required only in the root element inside a loop. We don't have to add it anywhere else. And remember - keys should be unique (if there are actual id's in the data - use that)

This snippet of code is wrong :

```jsx
<div key={someUniqueValue}>
  <h1 key={someUniqueValue}>Heading</h1>
  <p key={someUniqueValue}>Paragraph</p>
  <span key={someUniqueValue}>Span</span>
</div>
```

This snippet of code is correct :

```jsx
<div key={someUniqueValue}>
  <h1>Heading</h1>
  <p>Paragraph</p>
  <span>Span</span>
</div>
```

Considering this information, in your case we only have to add `key` attribute to `div` with class `album` :

```
<div className="album" key={item.id}>
  <Album item={item} />
</div>
```